### PR TITLE
Update DG

### PR DESCRIPTION
**Pull request by tanwartushar**
Update use cases, user stories

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -285,18 +285,27 @@ payment tracking and more in one efficient package, tailored specifically for fr
...
+| `* *`    | software developer | view clients on the blacklist                | view all the clients I intend to avoid                             |
+| `* *`    | software developer | whitelist a blacklisted client               | consider working with said client again                            |
+| `* *`    | software developer | view clients on the whitelist                | view all clients that are not on the blacklist                     |
+| `* *`    | software developer | archive a client                             | store information of old clients without cluttering my client list |
+| `* *`    | software developer | unarchive a client                           | bring the information of archived clients back to the client list  |
+| `* *`    | software developer | see reminders for the next earliest deadline | focus on the most urgent task first                                |
+| `*`      | software developer | keep notes for each client                   | take note of important details from meetings                       |
```
Comment (rahula1008) [inline docs/DeveloperGuide.md:304]: If we include this here, do we need to talk about this in the user guide under a coming soon section?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -306,68 +315,81 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
-* 1b. Input fields are incorrectly formatted
+* 1b. Input fields are incorrectly formatted.
 
     * 1b1. Clientele+ shows an error message.
-    * 1b2. User reenters command
-    Steps 1b1-1b2 are repeated until the command entered is correct
+    * 1b2. User reenters command. 
+
+      Steps 1b1-1b2 are repeated until the command entered is correct.
```
Comment (rahula1008) [inline docs/DeveloperGuide.md:337]: By any chance, would it be good to add the situation where at any time they decide to stop adding the client / exit the app?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -427,36 +469,58 @@ testers are expected to do more *exploratory* testing.
...
-   1. Re-launch the app by double-clicking the jar file.<br>
+   2. Re-launch the app by double-clicking the jar file.<br>
        Expected: The most recent window size and location is retained.
 
-1. _{ more test cases …​ }_
-
 ### Deleting a person
 
-1. Deleting a person while all persons are being shown
+1. Deleting a person while persons are being shown
```
Comment (rahula1008) [inline docs/DeveloperGuide.md:478]: For consistency, would it be good to replace 'persons' with clients here as well?

**Review by rahula1008 — COMMENTED** [review]
Few things to consider, but otherwise looks good

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (tanwartushar) [reply docs/DeveloperGuide.md:304]: I'll update the UG

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -427,36 +476,58 @@ testers are expected to do more *exploratory* testing.
...
+          1. Manually modify `addressbook.json`
+          2. Relaunch the app 
+          
+          Expected: Application fails to read from the file and loads without any data.
+       
+       2. Missing:
+          1. Move `data` folder to another directory.
+          2. Relaunch the app. 
+          
+          Expected: Application loads with last saved data.
```
Comment (jktang14) [inline docs/DeveloperGuide.md:533]: Is it last saved data? I thought if addressbook.json is missing its loaded with the sample address book

Comment (tanwartushar) [reply docs/DeveloperGuide.md:533]: yes thanks for pointing that out!

**Review by rahula1008 — APPROVED** [review]
LGTM

**Pull request merged**